plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -74,19 +75,26 @@ fu_plugin_dell_dock_probe(FuPlugin *plugin, FuDevice *proxy, GError **error)
 
 	/* create mst endpoint */
 	mst_device = fu_dell_dock_mst_new();
+	dev_guid = DELL_DOCK_VM5331_INSTANCE_ID;
+	if (!fwupd_guid_is_valid(dev_guid))
+		dev_guid = fwupd_guid_hash_string(dev_guid);
+	fu_device_set_context(FU_DEVICE(mst_device), ctx);
+	fu_device_add_guid(FU_DEVICE(mst_device), dev_guid);
 	fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(mst_device));
-	fu_device_add_instance_id(FU_DEVICE(mst_device), "MST-panamera-vmm5331-259");
 	if (!fu_plugin_dell_dock_create_node(plugin, FU_DEVICE(mst_device), error))
 		return FALSE;
 
 	/* create package version endpoint */
 	status_device = fu_dell_dock_status_new();
 	if (fu_dell_dock_module_is_usb4(FU_DEVICE(ec_device)))
-		instance = "USB\\VID_413C&PID_B06E&hub&salomon_mlk_status";
+		dev_guid = DELL_DOCK_DOCK2_INSTANCE_ID;
 	else
-		instance = "USB\\VID_413C&PID_B06E&hub&status";
+		dev_guid = DELL_DOCK_DOCK1_INSTANCE_ID;
+	if (!fwupd_guid_is_valid(dev_guid))
```

Comment:
same over here.  Maybe it makes sense to just set `dev_guid` to `fwupd_guid_hash_string(foo)` in both cases

Reply:
yes, will always true. Removed.  I added back the instance ID as it will be displayed in the client device list for better understanding of the relationship between GUID and sourced instance ID.